Initialize eviction restriction with pods controlled by VPA.

bskiba opened the pull request:
Additionally, remove failed and succeeded pods from live pods list.

bskiba commented:
@kgolab

kgolab commented on vertical-pod-autoscaler/pkg/updater/logic/updater.go:
```diff
@@ -111,6 +107,11 @@ func (u *updater) RunOnce() {
+	if u.evictionAdmission != nil {
+		u.evictionAdmission.LoopInit(controlledPods)
+	}
+	timer.ObserveStep("LoopInit")
```
It looks a bit strange now that we have "loop init" step inside the loop.
Since we'd probably want to keep it here maybe we can still rename the metric (AdmissionInit perhaps)?

bskiba commented on vertical-pod-autoscaler/pkg/updater/logic/updater.go:
```diff
@@ -152,16 +153,25 @@ func filterNonEvictablePods(pods []*apiv1.Pod, evictionRestriciton eviction.Pods
 }
 
+func isPodLive(pod *apiv1.Pod) bool {
+	if pod == nil {
```
I can remove, but I was planning to extract this method do common utils (I remember I have one more place I need to use this) and then it would make sense to keep the check.

bskiba commented:
Addressed the comments. Also, I noticed that the selector for podInformer already filters out Succeeded and Failed pods so I reverted to the filterDeletedPods method.

kgolab commented:
/LGTM
